libcodechecker/libclient/client.py

```diff
@@ -156,7 +148,93 @@ def setup_client(host, port, uri):
                       "login'.")
             sys.exit(1)
 
-    client = thrift_helper.ThriftClientHelper(host, port, uri, session_token)
+
+def setup_product_client(host, port, product_name=None, already_authed=False,
+                         session_token=None):
+    """
+    Setup the Thrift client for the product management endpoint.
+
+    :param already_authed  Don't check authentication, the caller guarantees
+      that a valid session exists.
+    """
+
+    if not already_authed:
+        # Check if the user has authenticated.
+        manager = session_manager.SessionManager_Client()
+        session_token = manager.getToken(host, port)
+        perform_auth_for_handler(manager, host, port, session_token)
+
+    if not product_name:
+        # Attach to the server-wide product service.
+        product_client = product_helper.ThriftProductHelper(
+            host, port, "/Products", session_token)
+    else:
+        # Attach to the product service and provide a product name
+        # as "viewpoint" from which the product service is called.
+        product_client = product_helper.ThriftProductHelper(
+            host, port, "/" + product_name + "/Products", session_token)
+
+        # However, in this case, the specified product might not be existing,
+        # which makes subsequent calls to this API crash (server sends
+        # HTTP 500 Internal Server Error error page).
+        if not product_client.getAPIVersion():
+            LOG.error("The product '{0}' cannot be communicated with. It "
+                      "either doesn't exist, or the server's configuration "
+                      "is bogus.".format(product_name))
+            sys.exit(1)
+
+    if not check_api_version(product_client):
+        LOG.critical("The server uses a newer version of the API which is "
+                     "incompatible with this client. Please update client.")
+        sys.exit(1)
+
+    return product_client
+
+
+def setup_client(product_url):
+    """
+    Setup the Thrift client and check API version and authentication needs.
+    """
+
+    try:
+        host, port, product_name = split_product_url(product_url)
+    except:
+        LOG.error("Couldn't understand the product URL!")
+        sys.exit(2)  # 2 for argument error.
+
+    # Check if the user has authenticated.
+    manager = session_manager.SessionManager_Client()
+    session_token = manager.getToken(host, port)
+    perform_auth_for_handler(manager, host, port, session_token)
+
+    # Check if the product exists.
+    product_client = setup_product_client(host, port,
+                                          product_name=None,
+                                          already_authed=True,
+                                          session_token=session_token)
+    product = product_client.getProducts(product_name, None)
+    product_error_str = None
+    if not (product and len(product) == 1):
+        product_error_str = "It does not exist."
```

Comment:
If you look at the context of this function, you'll see that the **full** error printed to the user is formatted as:

   The given product 'URL_ENDPOINT' cannot be used! {ERROR_REASON}

It would be superfluous to say "product" once again.